src/psyclone/dynamo0p3.py

```diff
@@ -1382,79 +1367,123 @@ def __init__(self, alg_calls):
         Invokes.__init__(self, alg_calls, DynInvoke)
 
 
-def stencil_extent_value(field):
-    '''Returns the content of the stencil extent. This may be a literal
-    value (a number) or a variable name. This function simplifies this
-    problem by returning a string in either case'''
-    if field.stencil.extent_arg.is_literal():
-        extent = field.stencil.extent_arg.text
-    else:
-        extent = field.stencil.extent_arg.varName
-    return extent
-
-
-def stencil_unique_str(arg, context):
-    '''Returns a string that uniquely identifies a stencil. As a stencil
-    differs due to the function space it operates on, type of
-    stencil and extent of stencil, we concatenate these things together
-    to return a unique string '''
-    unique = context
-    unique += arg.function_space.mangled_name
-    unique += arg.descriptor.stencil['type']
-    if arg.descriptor.stencil['extent']:
-        raise GenerationError(
-            "found a stencil with an extent specified in the metadata. This "
-            "is not coded for.")
-    unique += arg.stencil.extent_arg.text.lower()
-    if arg.descriptor.stencil['type'] == 'xory1d':
-        unique += arg.stencil.direction_arg.text.lower()
-    return unique
+class DynCollection(object):
```

Comment:
In the developer documentation please explain why you have a single abstract class dealing with invokes and kerns when you could have two. I think it is because sub-classes want to share information between the two so a single class allows this.

Reply:
Yes - otherwise all the variable declaration code would be duplicated. Sentence added to the docs.